src/main/java/org/reactome/addlinks/fileprocessors/HPAFileProcessor.java

```diff
@@ -0,0 +1,53 @@
+package org.reactome.addlinks.fileprocessors;
+
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class HPAFileProcessor extends FileProcessor<List<String>>
+{
+
+    public HPAFileProcessor(String processorName)
+    {
+        super(processorName);
+    }
+
+    public HPAFileProcessor()
+    {
+        super(null);
+    }
+
+    @Override
+    public Map<String, List<String>> getIdMappingsFromFile()
+    {
+        Map<String, List<String>> mappings = new HashMap<>();
+        try {
+            String dirToHPAFiles = this.unzipFile(this.pathToFile);
+            String inputFilepath = dirToHPAFiles + "/" + this.pathToFile.getFileName().toString().replace(".zip", "");
+
+            for (String line : Files.readAllLines(Paths.get(inputFilepath)).stream().skip(1).collect((Collectors.toList()))) {
+                List<String> tabSplit = Arrays.asList(line.split("\t"));
+
+                String geneName = tabSplit.get(0);
+                String ensemblId = tabSplit.get(2);
+                String uniprotId = tabSplit.get(4);
```

Comment:
Why is it that the indices accessed only the even-numbered ones?

Reply:
Cause we hate those odd-numbered whatsits! Just kidding, I think its actually just a coincidence

Reply:
Oh okay - what's in the odd-numbered indices?

Reply:
@jweiser Question not the reasoning behind the layout of the data files of others, for that way lies madness!!! :P

Reply:
@SolomonShorser-OICR I'll risk it - I must know the secrets of the data!

Reply:
This is the first 8 columns: `Gene | Gene synonym | Ensembl | Gene description | Uniprot | Chromosome | Position | Protein class', but there is actually over 200 columns in the file. Lots of raw data!

Reply:
(The | separator comes from me copying the file from Libre Office Calc (Ubuntu Excel)

Reply:
Please don't list all columns! Only the ones that are used. You could add a link to something online that describes all of the columns, if you really want to.

Reply:
I was just answering the questino for @jweiser lol, don't worry there is enough text in this PR already!

Reply:
Haha, could you just make the literal numbers named constants (e.g. `GENE_NAME_COLUMN`, `ENSEMBL_ID_COLUMN`, and `UNIPROT_ID_COLUMN`?